src/Uno.UWP/Graphics/Display/DisplayInformation.iOS.cs

```diff
@@ -171,30 +162,45 @@ private void UnobserveDisplayMetricsChanges()
 
 		static partial void SetOrientationPartial(DisplayOrientations orientations)
 		{
-			var currentOrientationMask = UIApplication.SharedApplication
-			   .StatusBarOrientation
-			   .ToUIInterfaceOrientationMask();
-
 			var toOrientationMask = orientations.ToUIInterfaceOrientationMask();
 
-			//If we are not already in one of the requested orientations, we need to force the application to rotate.
-			if (!toOrientationMask.HasFlag(currentOrientationMask))
-			{
-				//Rotate to the most preferred orientation that is requested
-				//e.g. if our mask is Portrait | PortraitUpsideDown, we prefer to initially rotate to Portrait rather than PortraitUpsideDown
-				var toOrientation = _preferredOrientations.FirstOrDefault(ori => toOrientationMask.HasFlag(ori)).ToUIInterfaceOrientation();
-
-				UIDevice.CurrentDevice
-					.SetValueForKey(
-						new NSNumber((int)toOrientation),
-						new NSString("orientation")
-					);
+			//Rotate to the most preferred orientation that is requested
+			//e.g. if our mask is Portrait | PortraitUpsideDown, we prefer to initially rotate to Portrait rather than PortraitUpsideDown
+			var toPreferredOrientationMask = _preferredOrientations.FirstOrDefault(ori => toOrientationMask.HasFlag(ori));
 
-				UIApplication.SharedApplication.SetStatusBarOrientation(toOrientation, false);
+			if (UIDevice.CurrentDevice.CheckSystemVersion(16, 0))
+			{
+				if (UIApplication.SharedApplication.KeyWindow is { } keyWindow
+					&& keyWindow.RootViewController is { } rootViewController
+					&& UIApplication.SharedApplication.ConnectedScenes.ToArray().FirstOrDefault() is UIWindowScene windowScene)
```

Comment:
Unlikely to happen but it would be better to have `First` throw (if set is empty) than having a NRE later with a null `windowScene`

Reply:
It seems to me that should not happen - the `if` will evaluate to false in case the `windowScene` is `null` so the remainder of the method will be skipped